src/components/codemirror.tsx

```diff
@@ -45,22 +58,156 @@ export default function CodeMirror(props: CodeMirrorProps) {
+				parent: parent.current,
+			})
+
+			setEditorRef(editor);
+			props.onEditorView?.(editor)
+			return
+		}
+
+		if(!props.roomState) return
+		if(!props.persistenceState) return
+		try{
+			if(yDoc !== undefined){
+				yDoc.destroy();
+			}
+			if(provider !== undefined){
+				provider.destroy();
+			}
+			yDoc = new Y.Doc();
+			provider = new WebrtcProvider(props.roomState.value.roomId, yDoc, {
+				signaling: [
+					"wss://yjs-signaling-server-5fb6d64b3314.herokuapp.com",
+				],
+			});
+			//get yjs document from provider
+			let ytext = yDoc.getText("codemirror");
+			const yUndoManager = new Y.UndoManager(ytext);
+
+			yProviderAwarenessSignal.value = provider.awareness
+			console.log(props.persistenceState.peek().session?.user)
+			const isHost = ((props.persistenceState.peek().kind == "PERSISTED" && props.persistenceState.peek().game != "LOADING") && props.persistenceState.peek().session?.user.id === props.persistenceState.peek().game.ownerId)
+			provider.awareness.setLocalStateField("user", {
+				name:
+					props.persistenceState.peek().session?.user.email ??
+					"Anonymous",
+				host: isHost
+			});
+			let yCollabExtension = yCollab(ytext, provider.awareness, {
+				undoManager: yUndoManager,
+			});
+			yCollabSignal.value = yCollabExtension;
+
+			//get the initial code from the yjs document
+			// Wait for document state to be received from provider
+			let initialUpdate = true;
+			const waitInitialUpdate = function () {
+				return new Promise<void>((resolve) => {
+					let timer: NodeJS.Timeout;
+					const checkUpdated = () => {
+						if (initialUpdate === false) {
+							console.log("Fjsakfjsa")
+							clearTimeout(timer);
+							resolve();
+						} else {
+							setTimeout(checkUpdated, 500);
+						}
+					};
+					timer = setTimeout(() => {
+						clearTimeout(timer);
+						resolve();
+					}, 1500);
+
+					checkUpdated();
+				});
+			};
+			waitInitialUpdate().then(() => {
+				if (ytext.toString() === "") {
+					ytext.insert(0, lastCode ?? "");
+				}
+				if (!parent.current)
+					throw new Error("Oh golly! The editor parent ref is null");
+				const editor = new EditorView({
+					state: createEditorState(ytext.toString(), () => {
+						if (editor.state.doc.toString() === lastCode) return
+						lastCode = editor.state.doc.toString()
+						onCodeChangeRef.current?.()
+					}, () => onRunShortcutRef.current?.(), yCollabSignal.peek() as Extension),
+					parent: parent.current,
+				})
+
+				setEditorRef(editor);
+				props.onEditorView?.(editor)
+			});
+			yDoc.on("update", () => {
+				if(!props.persistenceState) return;
+				console.log(provider.awareness.getStates());
+				if (!initialUpdate) return;
+				let participants: RoomParticipant[] = [];
+				provider.awareness.getStates().forEach((state) => {
+					try{
+						participants.push({
+							userEmail: state.user.name,
+							isHost: state.user.host
+						})
+					} catch(e){
+						return
```

Comment:
what happens if there's an error here?  should this actually blow up instead?

Reply:
it just means that the user won't be counted for. This doesn't happen for users, only for the special saving instance which doesn't have a name or host status, and it should just return and not count it